src/python_testing/conformance_support.py

```diff
@@ -50,13 +50,50 @@ def __str__(self):
         return f"ConformanceException({self.msg})"
 
 
+class ChoiceConformanceException(ConformanceException):
+    def __str__(self):
+        return f'ChoiceExceptions({self.msg})'
+
+
+class BasicConformanceException(ConformanceException):
+    pass
+
+
+@dataclass(frozen=True)
+class Choice:
+    choice: str
+    more: bool
+
+    def __str__(self):
+        more_str = '+' if self.more else ''
+        return '.' + self.choice + more_str
+
+
+def parse_choice(element: ElementTree.Element) -> Optional[Choice]:
+    choice = element.get('choice', '')
+    more = element.get('more', 'false') == 'true'
+    if choice and element.tag != OPTIONAL_CONFORM:
+        raise ChoiceConformanceException('Choice conformance on non-optional attribute')
+    return Choice(choice, more) if choice else None
+
+
 class ConformanceDecision(Enum):
     MANDATORY = auto()
     OPTIONAL = auto()
     NOT_APPLICABLE = auto()
     DISALLOWED = auto()
     PROVISIONAL = auto()
 
+    def set_choice(self, choice: Choice):
```

Comment:
Is this right? Enums are singletons, so if you do:

ConformanceDecision.MANDATORY.set_choice(123) then all of ConformanceDecision.MANDATORY have choice of 123.

I was going to suggest:

```py
def __init__(self, *a, **k):
   super().__init__(*a, **k)
   self.choice = None
```

to avoid the AttributeError below, however overall this feels off because of singletons. I believe you may need a dataclass with a Conformance member and a choice member. Enum does not seem made for having extra members.